op-supervisor/supervisor/backend/safety/views.go

```diff
@@ -0,0 +1,91 @@
+package safety
+
+import (
+	"errors"
+
+	"github.com/ethereum-optimism/optimism/op-service/eth"
+	"github.com/ethereum-optimism/optimism/op-supervisor/supervisor/backend/db/heads"
+	"github.com/ethereum-optimism/optimism/op-supervisor/supervisor/backend/db/logs"
+	"github.com/ethereum-optimism/optimism/op-supervisor/supervisor/types"
+)
+
+type View struct {
+	chainID types.ChainID
+
+	iter logs.Iterator
+
+	localView        heads.HeadPointer
+	localDerivedFrom eth.L1BlockRef
+
+	validWithinView func(l1View uint64, execMsg *types.ExecutingMessage) error
+}
+
+func (vi *View) Cross() (heads.HeadPointer, error) {
+	return vi.iter.HeadPointer()
+}
+
+func (vi *View) Local() (heads.HeadPointer, error) {
+	if vi.localView == (heads.HeadPointer{}) {
+		return heads.HeadPointer{}, logs.ErrFuture
+	}
+	return vi.localView, nil
+}
+
+func (vi *View) UpdateLocal(at eth.L1BlockRef, ref eth.L2BlockRef) error {
+	vi.localView = heads.HeadPointer{
+		LastSealedBlockHash: ref.Hash,
+		LastSealedBlockNum:  ref.Number,
+		//LastSealedTimestamp: ref.Time,
+		LogsSince: 0,
+	}
+	vi.localDerivedFrom = at
+
+	// TODO(devnet 2) reorg check against existing DB
```

Comment:
There's a bunch of TODOs here related to reorg checks, which we should reference the github issue in.